### PR TITLE
dotCMS/core#15769 SPAEM: Implement React Routing system

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -61,20 +63,25 @@ const server = http.createServer((request, response) => {
...
             // let postDataObject = JSON.parse(postData);
             const page = DotCMSApi.processPage(JSON.parse(parse(postData).dotPageData).entity);
-            
+
             fs.readFile(`${STATIC_FOLDER}/index.html`, 'utf8', (err, data) => {
-                const app = renderToString(<App data={page.layout} />);
+                const context = {};
+                const app = renderToString(
+                    <StaticRouter location={request.url} context={context}>
+                        <Page data={page.layout} />
```
Comment (freddyucv) [inline server/server.js]: why not put this into App?

File: src/App.js [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+    }
+
+    render() {
+        return <Page data={this.state.page.layout} />;
+    }
+}
+
+const App = data => {
+    return (
+        <Router>
```
Comment (freddyucv) [inline src/App.js]: what is the difference between Router and StaticRouter?

https://github.com/dotCMS/dotcms-spa-edit-mode/pull/6/files#diff-ea75b530d74f94670cea053feb8b961bR70

File: src/index.js [hunk 3]
```diff
@@ -2,20 +2,22 @@ import React from 'react';
...
+        </BrowserRouter>,
+        document.getElementById('root')
+    );
 } else {
-    DotCMSApi.getPage({
-        pathname: window.location.pathname
-    }).then(data => {
-        ReactDOM.render(<App data={data.layout} />, document.getElementById('root'));
-    })
+    ReactDOM.render(<App data={{just: 'testing'}} />, document.getElementById('root'));
```
Comment (alfredo-dotcms) [inline src/index.js]: ?

File: server/server.js (hunk 1 above)
Comment (fmontes) [reply server/server.js]: Becase `<App/>` now have a bunch of logic to handle the request to get get the page data and passing into `<Page/> in client mode but in server side we already have that data so we just pass it.

File: src/App.js (hunk 2 above)
Comment (fmontes) [reply src/App.js]: Router for client and StaticRouter for server.

File: src/index.js (hunk 3 above)
Comment (fmontes) [reply src/index.js]: Yes.

**Pull request merged**